src/Error/ExceptionRenderer.php

```diff
@@ -341,20 +344,33 @@ protected function _template(Throwable $exception, string $method, int $code): s
     }
 
     /**
-     * Get HTTP status code.
+     * Gets the appropriate http status code for exception.
      *
      * @param \Throwable $exception Exception.
-     * @return int A valid HTTP error status code.
+     * @return int A valid HTTP status code.
      */
-    protected function _code(Throwable $exception): int
+    protected function getHttpStatusCode(Throwable $exception): int
     {
-        $code = 500;
-        $errorCode = (int)$exception->getCode();
-        if ($errorCode >= 400 && $errorCode < 600) {
-            $code = $errorCode;
+        if ($exception instanceof HttpException) {
+            return (int)$exception->getCode();
         }
 
-        return $code;
+        // Controller exceptions
+        if ($exception instanceof MissingActionException) {
```

Comment:
It would be better to add a property which maintains a list of exceptions that should return a `404`. That would allow users to easily additional additional ones too as per their requirements.

As garas mentioned below `RecordNotFoundException` should also be in that list.

Reply:
Refs https://github.com/dereuromark/cakephp-tools/blob/master/src/Error/ErrorHandlerTrait.php#L36

Reply:
If we forsee people needing to map other codes too then we can use array format like
```php
[
    MissingActionException::class => 404,
    SomeOtherException::class => 400,
] 
```

Reply:
Done.